File: C++/Easy/1_Two_Sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int> &nums, int target) {
        vector<int> result;

        for(int i = 0; i < nums.size(); i++) {
            for(int j = i + 1; j < nums.size(); j++) {
                if(nums[i] + nums[j] == target) {
                    result.push_back(i);
                    result.push_back(j);
                    return result;
                }
            }
        }

        return result;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {3, 2, 4};
    int target = 6;

    vector<int> result = sol.twoSum(nums, target);

    for(int i = 0; i < result.size(); i++) {
        cout << result[i] << " ";
    }
    cout << endl;

    return 0;
}

File: Python/Easy/1_two_sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for index, num in enumerate(nums):
            complement = target - num
            # print(f"Index: {index}, Num: {num}, Complement: {complement}, Num Map: {num_map}")

            if complement in num_map:
                # print(f"Found: {num} + {complement} = {target}")
                # print(f"Indices: {num_map[complement]}, {index}")
                return [num_map[complement], index]
            num_map[num] = index
            # print(num_map[num])
        return []

# Example - you can change these numbers!
numbers = [2, 7, 11, 15]
target_value = 22

solver = Solution()
result = solver.twoSum(numbers, target_value)
print(f"The indices are: {result}")